### PR TITLE
Jenish/feature/iris-error-handling

### PR DESCRIPTION
**Pull request by JenishPatel99**

### CODE REVIEW

File: ex2_hal/iris/equipment_handler/source/iris_spi.c [hunk 1]
```diff
@@ -138,6 +138,8 @@ IrisLowLevelReturn iris_send_command(uint16_t command) {
     iris_nss_low();
     vTaskDelay(1);
     iris_spi_send(&command, 1);
+    vTaskDelay(1);
+    iris_spi_get(&rx_data, 1);
```
Comment (JenishPatel99) [inline ex2_hal/iris/equipment_handler/source/iris_spi.c:142]: This change was necessary for sending data to Iris

File: ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c [hunk 2]
```diff
@@ -97,14 +98,14 @@ int iris_write_page(uint32_t flash_addr, uint8_t *buffer) {
...
-    int ret = 0x00;
+    int ret = 0;
     uint8_t rx_data;
 
     /* First I2C transaction (2 bytes) */
     packet[0] = OPC_WRITE;
     packet[1] = N_OPC_WRITE;
-    iris_write_packet(packet, 2);
-    iris_read_packet(&rx_data, 1);
+    ret += iris_write_packet(packet, 2);
```
Comment (Taylorr82) [inline ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c:107]: There's no guarantee subsequent commands worked, shouldn't ret be checked every time and return on failure?

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: ex2_hal/iris/hardware_interface/source/iris.c [hunk 3]
```diff
@@ -121,7 +124,8 @@ Iris_HAL_return iris_take_pic() {
             break;
         }
         case FINISH: {
-            sys_log(INFO, "Iris returns ACK on take a picture command");
+            RTCMK_GetUnix(&time);
+            sys_log(INFO, "Iris commanded to take a picture at time: %d", time);
```
[the remark this answers is not in the document]
Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/source/iris.c:128]: Yes, fix has been made

File: ex2_hal/iris/hardware_interface/source/iris.c [hunk 4]
```diff
@@ -178,7 +183,7 @@ Iris_HAL_return iris_get_image_length(uint32_t *image_length) {
             break;
         }
         case FINISH: {
-            sys_log(INFO, "Iris returns ACK on get image length command");
+            sys_log(INFO, "Iris successfully returned image length");
```
[the remark this answers is not in the document]
Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/source/iris.c:186]: Good point, I have added that now

File: ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c (hunk 2 above)
Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c:107]: The granularity of error handling for Iris firmware updates is not required. We want to only capture if the overarching API fails or passes. Both `iris_write_packet` and `iris_read_packet` are wrappers for i2c handler, if one of them fails then it is guaranteed that both will fail. Also the subsequent commands are actually derived directly from the [STM32 manual ](file:///home/jenish/Downloads/an4221-i2c-protocol-used-in-the-stm32-bootloader-stmicroelectronics.pdf)

Comment (Taylorr82) [reply ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c:107]: That is fine, we don't need granularity for when it failed, but the OBC shouldn't continue doing stuff if it did fail, even if the stuff is harmless

Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c:108]: I guess that is right indeed. Have a look at the new changes

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**